package.json

```diff
@@ -2,17 +2,22 @@
   "name": "styled-components",
   "version": "2.3.0",
   "description": "Visual primitives for the component age. Use the best bits of ES6 and CSS to style your apps without stress 💅",
-  "main": "lib/index.js",
+  "main": "dist/styled-components.cjs.js",
   "typings": "typings/styled-components.d.ts",
   "jsnext:main": "dist/styled-components.es.js",
   "module": "dist/styled-components.es.js",
+  "react-native": "dist/styled-components.native.js",
   "scripts": {
-    "build": "npm run build:lib && npm run build:dist",
-    "prebuild:lib": "rimraf lib/*",
-    "build:lib": "babel --out-dir lib --ignore \"*.test.js\" src",
-    "prebuild:dist": "rimraf dist/*",
-    "build:dist": "rollup -c && rollup -c --environment ESBUNDLE && rollup -c --environment PRODUCTION",
-    "build:watch": "npm run build:lib -- --watch",
+    "build": "run-p build:**",
+    "prebuild": "rimraf dist",
+    "build:es": "cross-env ESBUNDLE=true rollup -c",
+    "build:cjs": "cross-env BABEL_ENV=cjs CJSBUNDLE=true rollup -c",
+    "build:umd:prod": "cross-env UMD=true PRODUCTION=true rollup -c",
+    "build:umd:dev": "cross-env UMD=true rollup -c",
+    "build:native": "cross-env BABEL_ENV=cjs NATIVEBUNDLE=true rollup -c",
+    "build:primitives": "cross-env BABEL_ENV=cjs PRIMITIVESBUNDLE=true rollup -c",
+    "build:no-parser:es": "cross-env NOPARSERBUNDLE=true rollup -c",
+    "build:no-parser:cjs": "cross-env BABEL_ENV=cjs NOPARSERBUNDLE=true rollup -c",
```

Comment:
Maybe let's move details to a new `scripts/build.js` and here provide only one task that we could call: 
`yarn build` to build all
`yarn build native` to build selected (passed by args)

Reply:
That's a great idea! it'd enable us to do both a general build, or a granular one 👍

Reply:
are granular builds really that useful and needed here? personally i liked the proposal of stuffing everything into a single rollup config (arrayed etc) - i guess though that with that in place it shouldnt be hard to introduce a possibility of granular builds by using some env variable like `BUILD_ONLY=granular_type`

Reply:
My suggestion is about moving implementation details out of `package.json` to keep it readable.